sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/UnevaluableAggs.scala

```diff
@@ -53,7 +53,6 @@ abstract class UnevaluableBooleanAggBase(arg: Expression)
   """,
   since = "3.0.0")
 case class BoolAnd(arg: Expression) extends UnevaluableBooleanAggBase(arg) {
```

Comment:
how about
```
case class BoolAnd(functionName: String, arg: Expression) ... with MultiNamesFunction {
  def nodeName = functionName
}
```

We can update `FunctionRegistry.expression` to detect `MultiNamesFunction` and pass name to the constructor.

Reply:
Updated PR. cc @cloud-fan